Framework/Algorithms/src/SampleCorrections/MCInteractionVolume.cpp

```diff
@@ -120,5 +162,40 @@ double MCInteractionVolume::calculateAbsorption(
          calculateAttenuation(afterScatter, lambdaAfter);
 }
 
+/**
+ * Generate a string summarising which parts of the environment
+ * the simulated scatter points occurred in
+ * @return The generated string
+ */
+std::string MCInteractionVolume::generateScatterPointStats() const {
+  std::stringstream scatterPointSummary;
+  scatterPointSummary << std::fixed;
+  scatterPointSummary << std::setprecision(2);
+
+  scatterPointSummary << "Scatter point counts:" << std::endl;
+
+  int totalScatterPoints = m_sampleScatterPoints;
+  for (std::vector<int>::size_type i = 0; i < m_envScatterPoints.size(); i++) {
+    totalScatterPoints += m_envScatterPoints[i];
```

Comment:
This loop can be replaced with
`totalScatterPoints += std::accumulate(m_envScatterPoints.begin(), m_envScatterPoints.end(), 0)`

Reply:
Done